docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md

```diff
@@ -10,6 +10,8 @@
 
 @endsphinxdirective
 
+> **NOTE**: This document contains description of Model Optimizer internals which should be used as a last resort. As internal API it may have changes which are not backward compatible in the future. If you are going to add support for operations from ONNX or PaddlePaddle, or you are not familiar with other extension alternatives in OpenVINO, please consider reading [this guide](../../Extensibility_UG/Intro.md) first.
```

Comment:
I think this statement is a bit too strong? If the frontend is *not* ONNX or PaddlePaddle (and TF?), then users have no choice to go down this route.

Reply:
So it is. I put it even more straight similar to how you formulated.

Reply:
For TF we don't have yet a new C++ based FE in stock, only as experimental feature that doesn't cover all currently supported TF models with old MO-based FE. So the user should follow this legacy/advanced stuff even for TF now.